controls/microsoft-365-foundations-1.2.2.rb

```diff
@@ -0,0 +1,46 @@
+        Users with permissions to the group mailbox can send as or send on behalf of the mailbox email address if the administrator has given that user permissions to do that. This is particularly useful for help and support mailboxes because users can send emails from "Contoso Support" or "Building A Reception Desk."
+        Shared mailboxes are created with a corresponding user account using a system generated password that is unknown at the time of creation.
+        The recommended state is Sign in blocked for Shared mailboxes.'
+    
+    desc 'check'
+    'Review Shared mailboxes in the UI:
+        1. Navigate to Microsoft 365 admin center https://admin.microsoft.com/
+        2. Click to expand Teams & groups and select Shared mailboxes.
+        3. Take note of all shared mailboxes.
+        4. Click to expand Users and select Active users.
+        5. Select a shared mailbox account to open its properties pane, and review.
+        6. Ensure the option reads Unblock sign-in.
+        7. Repeat for any additional shared mailboxes.
+    Note: If sign-in is not blocked it will read Block sign-in.
+    Audit in PowerShell using 2 modules:
+        1. Connect to Exchange Online using Connect-ExchangeOnline
+        2. Connect to Microsoft Graph using Connect-MgGraph -Scopes "Policy.Read.All"
+        3. Run the following PowerShell commands: 
+            $MBX = Get-EXOMailbox -RecipientTypeDetails SharedMailbox 
+            $MBX | ForEach-Object { Get-MgUser -UserId $_.ExternalDirectoryObjectId ` -Property DisplayName, UserPrincipalName, AccountEnabled } | Format-Table DisplayName, UserPrincipalName, AccountEnabled
+        4. Ensure AccountEnabled is set to False for all Shared Mailboxes.'
+    
+    desc 'fix'
+    "Block sign-in to shared mailboxes in the UI:
+        1. Navigate to Microsoft 365 admin center https://admin.microsoft.com/
+        2. Click to expand Teams & groups and select Shared mailboxes.
+        3. Take note of all shared mailboxes.
+        4. Click to expand Users and select Active users.
+        5. Select a shared mailbox account to open it's properties pane and then select Block sign-in.
+        6. Check the box for Block this user from signing in.
+        7. Repeat for any additional shared mailboxes."
+    'Remediate in PowerShell using 2 modules:
+        1. Connect to Microsoft Graph using Connect-MgGraph -Scopes "User.ReadWrite.All"
+        2. Connect to Exchange Online using Connect-ExchangeOnline.
+        3. To disable sign-in for a single account: $MBX = Get-EXOMailbox -Identity TestUser@example.com Update-MgUser -UserId $MBX.ExternalDirectoryObjectId -AccountEnabled:$false
+        3. The following will block sign-in to all Shared Mailboxes. $MBX = Get-EXOMailbox -RecipientTypeDetails SharedMailbox $MBX | ForEach-Object { Update-MgUser -UserId $_.ExternalDirectoryObjectId -AccountEnabled:$false }'
+    
+    impact 0.5
+    tag severity: 'medium'
+    
```

Comment:
probably a good idea to check other profiles to see what we do with cis controls that are 'explicitly not mapped'.  i think we still would want to have that metadata written out here.